src/cart/crt_iv.c

```diff
@@ -2755,6 +2755,13 @@ handle_response_cb(const struct crt_cb_info *cb_info)
 		info->cci_rpc = cb_info->cci_rpc;
 		info->cci_rc = cb_info->cci_rc;
 		info->cci_arg = cb_info->cci_arg;
+
+		/* handle timeouts on IV_FETCH synchronously to ensure unregister of bulk buffer
+		 * will occur before freeing it, in case emiter will finally make it
+		 */
+		if (cb_info->cci_rc == DER_TIMEDOUT && cb_info->cci_rpc->cr_opc == CRT_OPC_IV_FETCH)
```

Comment:
i think this should be -DER_TIMEDOUT, and as mentioned in the skype pm I think it would be safer to do != DER_SUCCESS instead as the timed out/unreachable node can result in few different errors depending on the rpc state such as -DER_CANCELLED, oog, unreachable and so on.

Reply:
Oops, right -DER_TIMEOUT for sure!!! Thx to have noticed it. Yes I was also thinking that more error cases/codes may be concerned, but I first wanted to expose this change to our specific scenario at Lenovo. By the way, I believe it should only be error codes that have been generated locally, like for the timeout, and not the ones generated from peer who will not attempt the RDMA/bulk in these cases.